schema/test/unit/application_validation_functions/carbon_taxed_fuels_match_rev_zero_test.sql

```diff
@@ -0,0 +1,324 @@
+      {
+        "application": ["_send_draft_application_email"],
+        "application_revision_status": ["_status_change_email"],
+        "ciip_user": ["_welcome_email"],
+        "form_result": ["_100_timestamps"]
+      }
+    ';
+    perform test_helper.modify_triggers('disable', disable_triggers);
+  end;
+$$;
+
+select test_helper.mock_open_window();
+select test_helper.create_test_users();
+select test_helper.create_applications(1, True, True);
+
+insert into ggircs_portal.application_revision(application_id, version_number) values (1, 0); -- creating an artificial revision zero with fuel form data
+insert into ggircs_portal.form_result(form_id, application_id, version_number, form_result)
+  values ((select id from ggircs_portal.form_json where slug='fuel'), 1, 0, '[]');
+
+-- test data
+insert into swrs.fuel_carbon_tax_details(normalized_fuel_type, carbon_tax_act_fuel_type_id)
+values ('Non carbon taxed test fuel - normalized', null),
+       ('Carbon taxed test fuel 1 - normalized', 1),
+       ('Carbon taxed test fuel 2 - normalized', 1); -- Aviation fuel
+
+insert into swrs.fuel_mapping(fuel_type, fuel_carbon_tax_details_id)
+values ('nCarbonTaxed', (select id from swrs.fuel_carbon_tax_details where normalized_fuel_type='Non carbon taxed test fuel - normalized')),
+       ('carbonTaxed1', (select id from swrs.fuel_carbon_tax_details where normalized_fuel_type='Carbon taxed test fuel 1 - normalized')),
+       ('carbonTaxed2', (select id from swrs.fuel_carbon_tax_details where normalized_fuel_type='Carbon taxed test fuel 2 - normalized'));
+
+insert into ggircs_portal.fuel(name, swrs_fuel_mapping_id)
+values ('nCarbonTaxed', (select id from swrs.fuel_mapping where fuel_type='nCarbonTaxed')),
+       ('carbonTaxed1', (select id from swrs.fuel_mapping where fuel_type='carbonTaxed1')),
+       ('carbonTaxed2', (select id from swrs.fuel_mapping where fuel_type='carbonTaxed2'));
+
+-- Should return true if there are no fuels
+update ggircs_portal.form_result set form_result = '[]'
+where application_id = 1 and form_id = (select id from ggircs_portal.form_json where slug='fuel');
+
+select is(
+  (
+    with app_revision as (
+      select row(application_revision.*)::ggircs_portal.application_revision
+      from ggircs_portal.application_revision
+      where application_id=1 and version_number=1
+    )
+    select ggircs_portal.carbon_taxed_fuels_match_rev_zero((select * from app_revision))
+  ),
+  true,
+  'Returns true if there are no fuels reported'
+);
+
+-- Should return true if the fuels match up
+update ggircs_portal.form_result set form_result =
+('[
+  {
+    "fuelRowId": '|| (select id from ggircs_portal.fuel where name = 'carbonTaxed1') || ',
+    "quantity": 10
+  },
+  {
+    "fuelRowId": '|| (select id from ggircs_portal.fuel where name = 'carbonTaxed2') || ',
+    "quantity": 100
+  }
+]')::jsonb where application_id = 1 and version_number = 0 and form_id = (select id from ggircs_portal.form_json where slug='fuel');
```

Comment:
FYI, you can make this a bit easier to read and robust with `json_build_array` and `json_build_object`